### DISCUSSION TITLE
How is the new moon conjunction defined?

### DISCUSSION THREAD

**Discussion by hidp123** [cosinekitty/astronomy#337] [Q&A]
Hi, I was wondering how the new moon conjunction is defined, in general, and in astronomy-engine specifically.

Looking online it is when the ecliptic longitude of the sun and moon are the same. However, what I was interested in is that is it that moment exactly or do we wait for a bit after the conjunction has taken place? And what is astronomy-engine's practice in regards to this?

**Comment by cosinekitty** [comment]
Yes, it is exactly as you say: the new moon is defined as the moment in time that the center of the Moon and the center of the Sun have the same ecliptic longitude as seen from the center of the Earth. Astronomy Engine finds this time without any adjustments. I hope this helps!

**Reply by Ahmed-Al-Imari** [reply]
Hi,
I need help in four small requests;
1- The crescent visibility code didn't open on my PC with MSYS2 power shell! Is there another way, or detailed instructions? Why it is not an executable file?
2- I want to show moon visibility map on my webpage that uses calculations in JavaScript: https://www.imamalmahdisigns.com/Original_Hijjri_Calendar/default.htm
3- I have a different moon visibility criteria to determine the lunar month first day that I want to develop this code using it to determine the first day of the lunar month. I am not a programmer and willing to pay for the new code.
4- I want to change the lunar month first day calculations in the above webpage to be based on moon visibility. The above webpage files will be provided. Willing to pay for this, too.
Have a nice day.
